From: Sven-Thorsten Dietrich <sdietrich@suse.de>
Subject: Use raw_smp_processor_id in __mem_cgroup_stat_add_safe

RT does not allow using smp_processor_id in preemptible code.

This needs to be reviewed for concurrency issues.

Signed-off-by: Sven-Thorsten Dietrich <sdietrich@suse.de>


---
 header          |    9 +++++++++
 mm/memcontrol.c |    2 +-
 2 files changed, 10 insertions(+), 1 deletion(-)

--- /dev/null
+++ b/header
@@ -0,0 +1,9 @@
+From: Sven-Thorsten Dietrich <sdietrich@suse.de>
+Subject: Use raw_smp_processor_id in __mem_cgroup_stat_add_safe
+
+RT does not allow using smp_processor_id in preemptible code.
+
+This needs to be reviewed for concurrency issues.
+
+Signed-off-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
+
--- a/mm/memcontrol.c
+++ b/mm/memcontrol.c
@@ -68,7 +68,7 @@ struct mem_cgroup_stat {
 static void __mem_cgroup_stat_add_safe(struct mem_cgroup_stat *stat,
 		enum mem_cgroup_stat_index idx, int val)
 {
-	int cpu = smp_processor_id();
+	int cpu = raw_smp_processor_id();
 	stat->cpustat[cpu].count[idx] += val;
 }
 
